pkg/qemu/qemu.go

```diff
@@ -422,6 +438,30 @@ func Cmdline(cfg Config) (string, []string, error) {
 	// Disk
 	baseDisk := filepath.Join(cfg.InstanceDir, filenames.BaseDisk)
 	diffDisk := filepath.Join(cfg.InstanceDir, filenames.DiffDisk)
+	extraDisks := []string{}
+	if len(y.AdditionalDisks) > 0 {
+		for _, diskName := range y.AdditionalDisks {
+			d, err := store.InspectDisk(diskName)
+			if err != nil {
+				logrus.Errorf("could not load disk %q: %q", diskName, err)
+				return "", nil, err
+			}
+
+			if d.Instance != "" {
+				logrus.Errorf("could not attach disk %q, in use by instance %q", diskName, d.Instance)
+				return "", nil, err
+			}
+			logrus.Infof("Mounting disk %q on %q", diskName, d.MountPoint)
+			err = d.Lock(cfg.InstanceDir)
```

Comment:
If there are any errors from here until `ha.Run()` is called, then we won't unlock the disk and the user will have to manually `limactl disk unlock DISK` before being able to use it again. It would be better to have some "unlock on error" handling somewhere in `hostagentAction`.